docs/proposals/HVPA-EP-0001/HVPA-EP-001.md

```diff
@@ -0,0 +1,803 @@
+The trade-off here is similar to the one in the case [multiple sources of resource recommendation](#multiple-sources-of-resource-recommendation).
+Hence, the proposed solution is also the same.
+
+Once the `desiredReplicas` are decided in this way, all the suggestions [mentioned](#scaling-only-vertically-or-horizontally-if-the-adjusted-recommended-resources-implies-vertical-scaling-per-pod-in-a-direction-opposite-to-that-of-the-total-resource-recommendation) [above](#base-resource-usage-per-replica) regarding scaling horizontally would apply to all the resources being scaled vertically.
+Again if the `desiredReplicas` decided in this way is equal to the `currentReplicas` then it would be a simple vertical scaling.
+
+#### Multiple containers in the target component pods
+
+If the target component contains multiple container in a pod, then `desiredReplicas` specified according to the matching [effective scaling interval](#effective-scaling-interval) might be different for the different containers.
+
+Such conflicts could be resolved in two ways.
+
+##### Sum of the recommendations for a resource for all the containers as total recommendation for the resource
+
+_Total resource recommendation_ for a resource could be redefined as the sum of the [total resource recommendation](#total-resource-recommendation) for that resource for all containers in the pod.
+All the other suggested approaches [above](#approach), would then be applied with this redefinition to decide the `desiredReplicas`.
+Once the `desiredReplicas` has been decided in this way, the adjusted resource recommendation per replica for each container would be calculated based on the original definition of [total resource recommendation](#total-resource-recommendation) for that container instead of the redefinition.
+
+##### Maximum of the `desiredReplicas` based on the total resource recommendation for each of the containers
+
+Another alternative is to treat this conflict in the same way as the conflict between the recommendations for [multiple resources](#multiple-resources) for the same container and extend the same solution.
+I.e. use the higher value of `desiredReplicas` among those specified for all the resources of all the container in the pod based on the [total resource recommendation](#total-resource-recommendation) for that resource for that container.
+Once the `desiredReplicas` has been decided in this way, the adjusted resource recommendation per replica for each resource and each container would be calculated based on the [total resource recommendation](#total-resource-recommendation) for that resource for that container.
+
+This approach is recommended in favour of [redefining total resource recommendation](#sum-of-the-recommendations-for-a-resource-for-all-the-containers-as-total-recommendation-for-the-resource).
```

Comment:
As discussed earlier, same set of `scalingIntervals` or buckets will be used to calculate `desiredReplicas` and the adjusted resource recommendation per replica for all the containers in a pod.
Do we also use the same `BaseResourcesPerReplica` too for all the containers when adjusting for the base resource usage?

Reply:
Yes, for the time-being. Unless the use-case for different configuration for different containers becomes strong.